web/src/test/java/com/replica/RouterServerTest.java

```diff
@@ -576,4 +629,81 @@ public void testPathDetailsReturnedOnlyWhenRequested() {
+        MultiPolygon tenMinInnerBucket = (MultiPolygon) wktReader.read(tenMinResponse.getBuckets(0).getGeometry());
+        assertTrue(fiveMinInnerBucket.getArea() < tenMinInnerBucket.getArea());
+
+        // With reverse flow set, buckets are different
+        MultiPolygon reverseFlowInnerBucket = (MultiPolygon) wktReader.read(reverseFlowResponse.getBuckets(0).getGeometry());
+        assertNotEquals(reverseFlowInnerBucket.getArea(), threeBucketInnerBucket.getArea());
+
+        assertThrows(RuntimeException.class, () -> {
+            routerStub.routeIsochrone(STREET_ISOCHRONE_REQUEST_OUT_OF_BOUNDS);
+        });
+    }
+
+    @Test
+    public void testPtIsochrone() throws ParseException {
+        final RouterOuterClass.IsochroneRouteReply threeBucketResponse = routerStub.routePtIsochrone(PT_ISOCHRONE_REQUEST_THREE_BUCKET);
+        final RouterOuterClass.IsochroneRouteReply fiveBucketResponse = routerStub.routePtIsochrone(PT_ISOCHRONE_REQUEST_FIVE_BUCKET);
+        final RouterOuterClass.IsochroneRouteReply fifteenMinResponse = routerStub.routePtIsochrone(PT_ISOCHRONE_REQUEST_FIFTEEN_MIN);
+        final RouterOuterClass.IsochroneRouteReply reverseFlowResponse = routerStub.routePtIsochrone(PT_ISOCHRONE_REQUEST_REVERSE_FLOW);
+
+        // Correct number of buckets returned
+        assertEquals(3, threeBucketResponse.getBucketsList().size());
+        assertEquals(3, fifteenMinResponse.getBucketsList().size());
+        assertEquals(5, fiveBucketResponse.getBucketsList().size());
+
+        // Bucket indices are correct
+        List<Integer> returnedBuckets = fiveBucketResponse.getBucketsList().stream()
+                .map(RouterOuterClass.IsochroneBucket::getBucket)
+                .collect(Collectors.toList());
+        assertEquals(List.of(0, 1, 2, 3, 4), returnedBuckets);
+
+        // Three-bucket/10 min response has larger buckets than five-bucket/10 min response
+        MultiPolygon threeBucketInnerBucket = (MultiPolygon) wktReader.read(threeBucketResponse.getBuckets(0).getGeometry());
+        MultiPolygon fiveBucketInnerBucket = (MultiPolygon) wktReader.read(fiveBucketResponse.getBuckets(0).getGeometry());
+        assertTrue(threeBucketInnerBucket.getArea() > fiveBucketInnerBucket.getArea());
+
+        // 10 min/3 bucket response has smaller buckets than 15 min/3 bucket response
+        MultiPolygon fiveMinInnerBucket = (MultiPolygon) wktReader.read(threeBucketResponse.getBuckets(0).getGeometry());
+        MultiPolygon fifteenMinInnerBucket = (MultiPolygon) wktReader.read(fifteenMinResponse.getBuckets(0).getGeometry());
+        assertTrue(fiveMinInnerBucket.getArea() < fifteenMinInnerBucket.getArea());
+
+        assertThrows(RuntimeException.class, () -> {
+            routerStub.routePtIsochrone(PT_ISOCHRONE_REQUEST_OUT_OF_BOUNDS);
+        });
+
+        // TODO: re-enable once reverseFlow works properly for PT isochrones
```

Comment:
@michaz while working on this test case, I found that the `reverseFlow` flag for transit isochrones doesn't seem to be working properly - no matter what point I tested with, the flag seemed to have no effect on the returned isochrone.

I see that in GraphHopper's master branch, the street-based isochrone endpoint has a [unit test for this flag](https://github.com/graphhopper/graphhopper/blob/master/web/src/test/java/com/graphhopper/application/resources/IsochroneResourceTest.java#L169), but there's no similar [test for the transit isochrone endpoint](https://github.com/graphhopper/graphhopper/blob/master/web/src/test/java/com/graphhopper/application/resources/PtIsochroneTest.java).

Could this perhaps be a bug? If that seems to be the case, we can unblock ourselves here in the short-term by having our server throw `NOT_IMPLEMENTED` or a similar error if someone sets this flag, to ensure there's no ambiguity about its (expected) behavior. But we'd love to have access to this feature in the future!

Reply:
As a short-term fix so we're not blocked, I've implemented the quick suggestion above^ (error if someone sets the reverse flow flag to true for transit isochrones) in [ac80d94](https://github.com/replicahq/graphhopper/pull/155/commits/ac80d94dd2db4c929d95d2f005fd7f613cad21d6)